cmd/file.d.go

```diff
@@ -90,18 +91,35 @@ func listenSignals() {
 	signal.Notify(signalChan, syscall.SIGHUP, syscall.SIGTERM)
 
 	for {
+		ctx, cancel := context.WithCancel(context.Background())
 		s := <-signalChan
 
 		switch s {
 		case syscall.SIGHUP:
 			logger.Infof("SIGHUP received")
-			fileD.Stop()
+
+			go func() {
+				cancel()
+			}()
+			err := fileD.Stop(ctx)
+			if err != nil {
+				logger.Fatalf("can't stop file.d with SIGHUP: %s", err.Error())
+			}
+
 			start()
 		case syscall.SIGINT:
 			fallthrough
```

Comment:
btw this doesn't make sense. it means SIGINT is ignored

Reply:
fallthrough means it is processed same way as case syscall.SIGTERM

i would personally prefer though 
case syscall.SIGINT, case syscall.SIGTERM:

in one case statement because the logic is the same

Reply:
if we want to catch SIGINT we need to add it on line 92 too